lib/circlex/api.ex

```diff
@@ -7,6 +7,9 @@ defmodule Circlex.Api do
 
   def auth(), do: Application.get_env(:circlex_api, :auth)
 
+  # Has to conform to the HTTPoison interface
+  def http_client(), do: Application.get_env(:circlex_api, :http_client, HTTPoison)
```

Comment:
Why not `@http_client Application.get_env(:circlex_api, :http_client, HTTPoison)`. This will then become a compile-time constant and thus allow a large number of compile-time optimizations (effectively erasure).